concepts/functions/about.md

```diff
@@ -6,60 +6,82 @@ Python, like other programming languages, has [_built-in functions_][build-in fu
+The function definition is followed by the function name and parentheses [`()`].
+Inside the parentheses, the parameters are specified, separated by commas.
+After the close parenthesis, the colon (`:`) is used to separate the function signature from the function body.
 
-The function body is a block of code that is executed when the function is called. The body of the function is indented. The indentation is important because Python relies on it to know where that block of code ends. A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
+The function body is a block of code that is executed when the function is called.
+The body of the function is indented.
+The indentation is important because Python relies on it to know where that block of code ends.
+A value can be returned from the function by using the `return` keyword, which can then be used by the caller of the function.
 
 ```python
 def function_name(parameter1, parameter2, ...):
     # function body
-    return value
+    return parameter1 + parameter2
+
 ```
 
 We can also define a function without any parameters or return value.
 
 ```python
 def function_name():
     # function body
+    pass
+
 ```
 
+Note that the function does need a body, even if the body does nothing, or trying to run the program will generate an indentation error.
+
 ## Calling a Function
 
-To call a function, use the function name followed by parenthesis [`()`]. Parameters passed to the function are placed inside the parenthesis, separated by commas.
+To call a function, use the function name followed by parenthesese [`()`].
+Parameters passed to the function are placed inside the parenthesese, separated by commas.
 
 Consider the following function:
 
 ```python
-def wish():
+def greet():
     print("Hello")
+
 ```
 
 The above function can be called by using the following syntax:
 
 ```python
->>> wish()
+>>> greet()
 Hello
 ```
 
-## Parameters
+## Parameters and their Arguments
 
-Parameters are values that are passed to the function when it is called. They can be of any data type.
+Arguments are values that are passed to the function when it is called.
+They can be of any data type.
 
 Let's define a function `add` which adds two numbers together:
 
 ```python
-def add(x, y):
-    print(x + y)
+def add(first, second):
+    print(first + second)
+
 ```
 
-When the function is called, the parameters are passed to the function. We need to pass values for both the parameters, otherwise a [`TypeError`][type-error] will be raised.
+The parameters `first` and `second` define what arguments the `add` function will accept.
+(It should be noted that the words `parameter` and `argument` are often used interchangeably, albeit imprecisely.)
+When the function is called, the arguments are passed to the function.
+We need to pass arguments for both of the parameters, otherwise a [`TypeError`][type-error] will be raised.
```

Comment:
```suggestion
Because the function definition outlines two parameters, arguments for both of the parameters will need to be passed on function call - otherwise a [`TypeError`][type-error] will be raised:
```